src/NodaTime.Demo/PeriodDemo.cs

```diff
@@ -86,6 +86,146 @@ public void ConstructionFromNanoseconds()
             Period period = Snippet.For(Period.FromNanoseconds(42));
             Assert.AreEqual(42, period.Nanoseconds);
             Assert.AreEqual("PT42n", period.ToString());
-        }        
+        }
+
+        [Test]
+        public void Between_LocalDatesDaysApart()
+        {
+            Period period = Snippet.For(Period.Between(
+                new LocalDate(2017, 11, 10),
+                new LocalDate(2017, 11, 15)));
+
+            Assert.Multiple(() =>
+            {
+                Assert.AreEqual(5, period.Days);
+                Assert.AreEqual("P5D", period.ToString());
+            });
+        }
+
+        [Test]
+        public void Between_LocalDatesYearsApart()
+        {
+            Period period = Snippet.For(Period.Between(
+                new LocalDate(1990, 6, 26),
+                new LocalDate(2017, 11, 15)));
+
+            Assert.Multiple(() =>
+            {
+                Assert.AreEqual(27, period.Years);
+                Assert.AreEqual(4, period.Months);
+                Assert.AreEqual(20, period.Days);
+                Assert.AreEqual("P27Y4M20D", period.ToString());
+            });
+        }
+
+        [TestCase(PeriodUnits.Months, 5, 0, 0, "P5M")]
+        [TestCase(PeriodUnits.Weeks, 0, 22, 0, "P22W")]
+        [TestCase(PeriodUnits.Months | PeriodUnits.Weeks, 5, 1, 0, "P5M1W")]
+        [TestCase(PeriodUnits.Months | PeriodUnits.Days, 5, 0, 7, "P5M7D")]
+        [TestCase(PeriodUnits.Days, 0, 0, 160, "P160D")]
+        public void Between_LocalDatesWithGivenUnits(PeriodUnits units, int expectedMonths, int expectedWeeks, int expectedDays, string expectedPattern)
```

Comment:
This would be great as a regular unit test, but won't work for snippets. It would be better to just choose the most interesting example - maybe a "years and days" one, as something a bit different?